proxies/remote-proxy-legacy/apiproxy/resources/jsc/set-jwt-variables.js

```diff
@@ -44,7 +42,7 @@ try {
         });
     }
     
-    if (isValidApiKey) {
+    if (apiProductsList.length > 0) {
```

Comment:
Even when there is no approved product, this is somehow a list with one element (empty string `""`) and `isValidKey` will be set to `true`.

Not verified but the same likely applies to the gcp proxies.

Reply:
I cannot recreate. Also, looking at the code, it appears to me that the only way this can happen would be if the apiProduct is approved, valid, and has an apiProduct.Name = "". Can you recheck your setup?